File: Problem53/Solution.go
package Problem53

import (
	"fmt", "math"
)

func maxSubArray(nums []int) int {
    max := nums[0];
    localMax := nums[0];
    for i:=1; i < len(nums); i++ {
    	if localMax + nums[i] > nums[i] {
	    	localMax = localMax + nums[i];
    	} else {
    		localMax = nums[i]
    	}
    	
    	if localMax > max {
    		max = localMax;
    	}
    	
    }
    return max;
}

File: Problem53/Solution.java
public class Solution {
	public int maxSubArray(int[] nums) {
		int max = nums[0], localSum = nums[0];
		for (int i = 1; i < nums.length; i++) {
			localSum = Math.max(localSum + nums[i], nums[i]);
			max = Math.max(max, localSum);
		}
		return max;
	}
}